rasterio/rio/edit_info.py

```diff
@@ -58,7 +59,10 @@ def transform_handler(ctx, param, value):
         except ValueError:
             pass
         try:
-            retval = guard_transform(value)
+            if tastes_like_gdal(value):
```

Comment:
@brendan-ward You were right about having `guard_transform()` raise an exception for a period of time.  The test ensuring that GDAL style transforms can be passed to `$ rio edit-info --transform ${VALUE}` _should_ have started failing along with some of the other `affine vs. transform` changes, but _didn't_ until I added the exception to `guard_transform()`.  I suspect some of our users will hit the same problem.

  Passing a GDAL style transform into `$ rio edit-info` seems useful, but I didn't notice one of the related tests was misbehaving until I added an exception to `guard_transform()`.